modules/lib/core/index.d.ts

```diff
@@ -88,7 +147,7 @@ export interface Content<
     originProject?: string;
     childOrder?: string;
     _sort?: object[];
-    page: Type extends 'portal:fragment' ? undefined : _Component;
+    page: Type extends 'portal:fragment' ? undefined : Page;
```

Comment:
We are missing something here...

This would not give a union of all Page-types automatically.

We would need something like:

```typescript
type EveryPageDescriptor = keyof XpPartMap
type EveryPage = EveryPageDescriptor extends any // this lets us iterate over every member of the union
 ? Part<EveryPageDescriptor>
 : never;
```

Here comes this line:
```suggestion
    page: Type extends 'portal:fragment' ? undefined : EveryPage;
```

Reply:
Also one looses the ability to specify a specific page, so perhaps removing the third generic param is a mistake

Reply:
Yeah I think we are mixing to extremes, instead of providing both options...